recipes-openxt/openxt/openxt-keymanagement/key-functions

```diff
@@ -0,0 +1,551 @@
+                hide_cursor
+            else
+                key_file=${key_file:-"${KEYDIR}/recovery"}
+                echo -n ${well_known} > ${key_file}
+            fi
+        else
+            key_file=$(password_prompt "device recovery" "recovery")
+        fi
+    fi
+
+    key_file=${key_file:-"${KEYDIR}/recovery"}
+
+    if [ ! -e ${key_file} ]; then
+        echo -n ${well_known} > ${key_file}
+    fi
+
+    chcon -t lvm_tmp_t ${key_file}
+    echo ${key_file}
+}
+
+gen_platform_key() {
+    local key_file=`mktemp -t`
+
+    get-config-key > ${key_file}
+
+    chcon -t lvm_tmp_t ${key_file}
+    echo $key_file
+}
+
+# Generates the config partition key
+gen_config_key() {
+    local root="${1:-/}"
+    local key_file="${root}/config/keys/config.key"
+
+    seed_entropy 2>/dev/null
+    dd if=/dev/random bs=1 count=32 of="${key_file}" 2>/dev/null || return 1
+
+    echo $key_file
+}
+
+# Generates the log partition key
+gen_log_key() {
+    local root="${1:-/}"
+    local key_file="${root}/config/keys/log.key"
+
+    seed_entropy 2>/dev/null
+    dd if=/dev/urandom bs=1 count=128 of="${key_file}" 2>/dev/null || return 1
+
+    echo $key_file
+}
+
+# Generates the cores partition key
+gen_cores_key() {
+    local root="${1:-/}"
+    local key_file="${root}/config/keys/cores.key"
+
+    seed_entropy 2>/dev/null
+    dd if=/dev/urandom bs=1 count=128 of="${key_file}" 2>/dev/null || return 1
+
+    echo $key_file
+}
+
+# Generates the device binding key
+gen_device_key() {
+    local root="${1:-/}"
+    local key_file="${root}/config/sec/device.key"
+
+    seed_entropy 2>/dev/null
+    dd if=/dev/urandom bs=1 count=128 of="${key_file}" 2>/dev/null || return 1
+
+    echo $key_file
+}
+
+# Generates the TPM own key/password
+get_own_key() {
+    local key_file=""
+
+    key_file=$(password_prompt "TPM Own")
+
+    echo ${key_file}
+}
+
+# Gets the recovery key/password
+get_recovery_key() {
+    local key_file=""
+
+    key_file=$(password_prompt "device recovery")
+
+    echo ${key_file}
+}
+
+get_platform_key() {
+    local key_file=`mktemp -t`
+
+    get-config-key > ${key_file}
+
+    echo $key_file
+}
+
+# Gets the config partition key
+get_config_key() {
+    local root="${1:-/}"
+    local key_file="${root}/config/keys/config.key"
+
+    if [ -e ${key_file} ]; then
+        echo $key_file
+        return 0
+    else
+        echo ""
+        return 1
+    fi
+}
+
+# Gets the log partition key
+get_log_key() {
+    local root="${1:-/}"
+    local key_file="${root}/config/keys/log.key"
+
+    if [ -e ${key_file} ]; then
+        echo $key_file
+        return 0
+    else
+        echo ""
+        return 1
+    fi
+}
+
+# Gets the cores partition key
+get_cores_key() {
+    local root="${1:-/}"
+    local key_file="${root}/config/keys/cores.key"
+
+    if [ -e ${key_file} ]; then
+        echo $key_file
+        return 0
+    else
+        echo ""
+        return 1
+    fi
+}
+
+# Gets the device binding key
```

Comment:
A simple if logic that is 7 lines in total is not much duplication, it would be one thing if the refactor increased readability/logic reasoning but this refactor would result in nothing more than a does_file_exist function. Now if the logic was more complex like make sure the file existed, was readable and hand a fingerprint of xyz, then IMHO that would be worthy of a refactor.